earth_enterprise/src/fusion/autoingest/plugins.src/MercatorRasterProject.src

```diff
@@ -942,19 +940,15 @@ uint32 MercatorRasterProjectAssetVersionImplD::BuildIndex(
     // Determine which insets (0 .. i) intersect this coverage area.
     std::vector<uint> neededIndexes;
     neededIndexes.reserve(i);
-    if (type == AssetDefs::Imagery) {
-      FindNeededImageryInsets(genInfo[i].coverage,
-                              insetInfoPtrs, i,
-                              neededIndexes,
-                              beginMinifyLevel,
-                              endMinifyLevel);
-    } else {
-      FindNeededTerrainInsets(genInfo[i].coverage,
-                              insetInfoPtrs, i,
-                              neededIndexes,
-                              beginMinifyLevel,
-                              endMinifyLevel);
-    }
+
+    overlapEnvelope<MercatorRasterProductAssetVersion> env (
+        type, genInfo[i].coverage, insetInfoPtrs, i,
+        beginMinifyLevel, endMinifyLevel, i);
+
+    std::shared_ptr<OverlapCalculator<MercatorRasterProductAssetVersion>> calc =
```

Comment:
Minor: thinking about this a bit more: do you need a shared_ptr at all here? I'm not sure what the shared_ptr is getting us since we don't have lots of parts of the code using this over and over. I suspect just passing around the object (rather than a shared pointer to the object) would work fine.